client/blocks/reader-suggested-follows/dialog.jsx

```diff
@@ -1,71 +1,119 @@
 import { Dialog } from '@automattic/components';
 import { useTranslate } from 'i18n-calypso';
 import { useEffect } from 'react';
-import SuggestedFollowItem from 'calypso/blocks/reader-suggested-follows';
-import { useRelatedSites } from 'calypso/data/reader/use-related-sites';
+import { SuggestedFollowItem } from 'calypso/blocks/reader-suggested-follows';
 import { READER_SUGGESTED_FOLLOWS_DIALOG } from 'calypso/reader/follow-sources';
 import { useDispatch } from 'calypso/state';
 import { recordReaderTracksEvent } from 'calypso/state/reader/analytics/actions';
-
+import { useRecommendOrRelatedSitesQuery } from './hooks/use-recommend-or-related-sites-query';
+import { RecommendedFeed } from './recommended-feed';
 import './style.scss';
 
-const ReaderSuggestedFollowsDialog = ( { onClose, siteId, postId, isVisible } ) => {
+const ReaderSuggestedFollowsDialog = ( { onClose, siteId, postId, isVisible, author = {} } ) => {
 	const dispatch = useDispatch();
 	const translate = useTranslate();
-	const { data, isLoading } = useRelatedSites( siteId, postId );
+	const userName = author?.name;
+
+	const { data, isLoading, resourceType } = useRecommendOrRelatedSitesQuery(
+		{
+			author,
+			siteId,
+			postId,
+		},
+		{
+			enabled: isVisible,
+		}
+	);
 
 	useEffect( () => {
-		if ( isVisible ) {
-			dispatch( recordReaderTracksEvent( 'calypso_reader_suggested_follows_dialog_viewed' ) );
+		if ( isVisible && resourceType ) {
+			dispatch(
+				recordReaderTracksEvent( 'calypso_reader_suggested_follows_dialog_viewed', {
+					resource_type: resourceType,
+				} )
+			);
 		}
-	}, [ isVisible, dispatch ] );
+	}, [ isVisible, dispatch, resourceType ] );
 
 	// If we are no longer loading and no data available, don't show the dialog
-	if ( ! isLoading && data === undefined ) {
+	if ( ! isLoading && resourceType === null ) {
 		return null;
 	}
 
+	const title =
+		resourceType === 'recommended'
+			? translate( 'Recommended sites' )
+			: translate( 'Suggested sites' );
+
+	const description =
+		resourceType === 'recommended'
+			? translate( 'While you’re at it, check out these sites %(authorName)s recommends.', {
+					args: { authorName: userName },
+			  } )
+			: translate( 'While you’re at it, you might check out these sites.' );
+
```

Comment:
Let's use recommended "blogs" here to match the other places in the Reader


```suggestion
			? translate( 'Recommended blogs' )
			: translate( 'Suggested sites' );

	const description =
		resourceType === 'recommended'
			? translate( 'While you’re at it, check out these blogs %(authorName)s recommends.', {
					args: { authorName: userName },
			  } )
			: translate( 'While you’re at it, you might check out these sites.' );

```

Reply:
Updated here [2de585f](https://github.com/Automattic/wp-calypso/pull/104927/commits/2de585fa043cd6b33b1faa15518dc98be858a5fa)